src/main/java/spleetwaise/transaction/logic/commands/FilterCommand.java

```diff
@@ -0,0 +1,131 @@
+package spleetwaise.transaction.logic.commands;
+
+import static spleetwaise.address.logic.parser.CliSyntax.PREFIX_PHONE;
+import static spleetwaise.transaction.logic.parser.CliSyntax.PREFIX_AMOUNT;
+import static spleetwaise.transaction.logic.parser.CliSyntax.PREFIX_DATE;
+import static spleetwaise.transaction.logic.parser.CliSyntax.PREFIX_DESCRIPTION;
+
+import java.util.function.Predicate;
+
+import spleetwaise.address.commons.util.ToStringBuilder;
+import spleetwaise.address.logic.commands.CommandResult;
+import spleetwaise.address.model.person.Person;
+import spleetwaise.commons.CommonModel;
+import spleetwaise.transaction.logic.commands.exceptions.CommandException;
+import spleetwaise.transaction.model.transaction.Amount;
+import spleetwaise.transaction.model.transaction.Date;
+import spleetwaise.transaction.model.transaction.Description;
+import spleetwaise.transaction.model.transaction.Transaction;
+
+/**
+ * This class represents a command for filtering transactions.
+ */
+public class FilterCommand extends Command {
+
+    /**
+     * The word used to trigger this command in the input.
+     */
+    public static final String COMMAND_WORD = "filterTxn";
+
+    /**
+     * The message that is displayed upon successful execution of this command.
+     */
+    public static final String MESSAGE_SUCCESS = "Transaction book filtered.";
+
+    /**
+     * The message usage string that explains how to use this command.
+     */
+    public static final String MESSAGE_USAGE =
+            COMMAND_WORD + ": Filter the transaction book.\n" + "Parameters: "
+                    + "[" + PREFIX_PHONE + "CONTACT] " + "[" + PREFIX_AMOUNT + "AMOUNT] " + "[" + PREFIX_DESCRIPTION
+                    + "DESCRIPTION] " + "[" + PREFIX_DATE + "DATE ]\n"
+                    + "At least one of the above filtering criteria is needed\n"
+                    + "Example: " + COMMAND_WORD + " " + PREFIX_PHONE + "88888888";
+
+    private final Person contact;
+    private final Amount amount;
+    private final Description description;
+    private final Date date;
+
+    /**
+     * Creates an AddCommand to add the specified {@code Transaction}.
+     *
+     * @param contact The person to filter the transaction by.
+     * @param amount The amount to filter the transaction by.
+     * @param description The description to filter the transaction by.
+     * @param date The date to filter the transaction by.
+     */
+    public FilterCommand(Person contact, Amount amount, Description description, Date date) {
```

Comment:
Is it possible to follow how FindCommand.java is implemented?

with a `private final TxnContainsKeywordsPredicate predicate;`